generator/src/main/java/com/graphicsfuzz/generator/tool/KnownValueShaderGenerator.java

```diff
@@ -152,7 +154,11 @@ public static void mainHelper(String[] args) throws ArgumentParserException, IOE
         new ExprStmt(new BinaryExpr(new VariableIdentifierExpr("_GLF_color"),
             new TypeConstructorExpr("vec4", rgbaExprs), BinOp.ASSIGN)));
 
-    final ShaderJob newShaderJob = new GlslShaderJob(Optional.empty(), new PipelineInfo(), tu);
+    final ShaderJob newShaderJob = new GlslShaderJob(Optional.empty(), pipelineInfo, tu);
+    // We randomly choose some of the uniforms generated by the generator to be pruned.
+    PruneUniforms.prune(newShaderJob, generator.nextInt(pipelineInfo.getNumUniforms()),
```

Comment:
Can you make --max-uniforms be a command-line option, and only prune uniforms if it is set?  (In which case prune to the given value.)

Also, can you make --generate-uniform-bindings a command line option, and make it do what it does in Generate?  (This is important for Vulkan.)

Reply:
Ah, I am now understanding how those arguments are related to the pruning. Thanks for your helpful feedback :-)